File: dsa-problems/Probabilistic/solutions/javascript/PRB-009-treap-priority-invariants.js
const readline = require("readline");

function solve(n) {
  let h = 0.0;
  for (let i = 1; i <= n; i++) {
    h += 1.0 / i;
  }
  return h;
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let data = [];
rl.on("line", (line) => data.push(line.trim()));
rl.on("close", () => {
  if (data.length === 0) return;
  const n = parseInt(data[0], 10);
  console.log(solve(n).toFixed(6));
});

File: dsa-problems/Probabilistic/solutions/python/PRB-009-treap-priority-invariants.py
import sys

def treap_expectations(n: int):
    H = 0.0
    for i in range(1, n + 1):
        H += 1.0 / i

    return H

def main():
    input = sys.stdin.read
    data = input().split()
    if not data:
        return
    n = int(data[0])
    h = treap_expectations(n)
    print(f"{h:.6f}")

if __name__ == "__main__":
    main()
